gson/src/org/immutables/gson/stream/JsonGeneratorWriter.java

```diff
@@ -142,9 +142,8 @@ public JsonWriter value(Number value) throws IOException {
     if (value == null) {
       return nullValue();
     }
-    double d = value.doubleValue();
     if (!isLenient()) {
-      if (Double.isNaN(d) || Double.isInfinite(d)) {
+      if (Double.isNaN(value.doubleValue()) || Double.isInfinite(value.doubleValue())) {
```

Comment:
this could be a silly comment on my side, I've noticed that we call `value.doubleValue()` many times, cannot justify the impact because of jit/inlining, just mentioning

Reply:
We'll call it only once in `else`.
See my [previous comment](https://github.com/immutables/immutables/pull/1120#discussion_r345316915)

Reply:
When strict parsing is enabled, You will have to pay the cost of either a cast, or conversion to double for all Number types that can be NaN or infinite. I'm open to how you want to do that, but can't  speculate on what the JVM does performance wise.

Reply:
Why not check only for Floats and Doubles (BigDecimal is a special case) when strict checking is enabled ? 

```java
    System.out.printf("POSITIVE_INFINITY: %s%n", Double.isInfinite(Float.POSITIVE_INFINITY));
    System.out.printf("NEGATIVE_INFINITY: %s%n", Double.isInfinite(Float.NEGATIVE_INFINITY));
    System.out.printf("NaN: %s%n", Double.isNaN(Float.NaN));
```

> POSITIVE_INFINITY: true
> NEGATIVE_INFINITY: true
> NaN: true

Reply:
I would say have private method:

```java

public JsonWriter value(Number value) {

   if (value instanceof Float) {
     float f = value.floatValue();
     checkStrictNumber(f);
     generator.writeNumber(f);	
   } else {
      double d = value.doubleValue();
      checkStrictNumber(d);
      generator.writeNumber(d);	
   }
}

void checkStrictNumber(double number) {
  if (isLinient() && (Double.isInfinite(number) || Double.isNaN(number))) {
    throw new IllegalArgumentException("JSON forbids NaN and infinities: " + value);
  }
} 
```

Do you agree with this approach ?

Reply:
Yep will implement.